apps/aot/src/utils/time-utils.ts

```diff
@@ -1,35 +1,38 @@
-export function hasAdventStarted() {
-  const today = new Date().getTime();
-  const aotStartTime = new Date(Date.UTC(2024, 11, 1, 5, 0, 0));
+export function hasAdventStarted(year: number) {
+  if (!isValidAdventYear(year)) return false;
 
-  return today >= aotStartTime.getTime();
+  const now = new Date().getTime();
+  const aotStartTime = new Date(Date.UTC(year, 11, 1, 5, 0, 0));
+
+  return now >= aotStartTime.getTime();
 }
 
-export function getCurrentAdventDay() {
+export function getCurrentAdventDay(year: number) {
   const MS_PER_DAY = 24 * 60 * 60 * 1000;
 
-  const today = new Date().getTime();
-  const startDate = new Date('2024-12-01T05:00:00.000Z').getTime();
-  const endDate = new Date('2024-12-26T05:00:00.000Z').getTime();
+  const now = new Date().getTime();
+  const startDate = new Date(Date.UTC(year, 11, 1, 5, 0, 0)).getTime();
+  const endDate = new Date(Date.UTC(year, 11, 26, 5, 0, 0)).getTime();
 
-  if (today < startDate) {
+  if (now < startDate) {
     return 0; // not advent yet!
   }
 
-  if (today > endDate) {
+  if (now > endDate) {
     return 25;
   }
 
-  const adventDay = Math.floor((today - startDate) / MS_PER_DAY) + 1;
+  const adventDay = Math.floor((now - startDate) / MS_PER_DAY) + 1;
 
   return adventDay;
 }
 
 export function isValidAdventDay(selectedDay: number) {
-  if (selectedDay < 1 || selectedDay > 25) return false;
+  return selectedDay >= 1 && selectedDay <= 25;
+}
 
-  const currentAdventDay = getCurrentAdventDay();
-  return selectedDay <= currentAdventDay;
+export function isValidAdventYear(selectedYear: number) {
+  return selectedYear >= 2023 && selectedYear <= 2024;
```

Comment:
how would this work. would we need to keep adjust thing this manually?

Reply:
Whatever is easiest tbh, maybe we can extract to some const variables somewhere? Or you can do something similar to where you queried the DB for the events and we can make sure it is part of that, I just wanted to avoid querying the db to do a simple validation.

Reply:
its good enough ™ for now